modules/system/checks.nix

```
@@ -44,6 +44,18 @@ let
     fi
   '';
 
+  checkHomebrewInstall = ''
+    if [ ! -f /usr/local/bin/brew ] && [ ! -f /opt/homebrew/bin/brew ]; then
```

For the case when someone is using a custom directory for brew

https://stackoverflow.com/a/677212
```suggestion
    if [ command -v brew ]; then
```

---

The thing is, this does not work. The `brew` binary is not in path by default. At the end of the homebrew installation, we're asked to put the following line `eval "$(/usr/local/bin/brew shellenv)"` in our `.zprofile` or `.bashrc`. This is what loads `brew` into the path. Ergo, using something like `command -v brew` will always fail when nix is running since it doesn't evaluate the user's zsh / bash profile.

---

But I do recognize my solution would fail for someone using a custom directory for brew. Not sure what an optimal solution for that case should be. We cannot even `eval "$(/usr/local/bin/brew shellenv)"` in the nix code, since that too would require knowledge about `brew`'s installation path.

---

This should use `${config.homebrew.brewPrefix}/brew`, like the Homebrew module itself does.

---

```suggestion
    if [ ! -f ${config.homebrew.brewPrefix}/brew ]; then
```